lisa/tools/kill.py

```diff
@@ -40,12 +41,15 @@ def by_name(self, process_name: str, signum: int = SIGKILL) -> None:
             )
 
     def by_pid(self, pid: str, signum: int = SIGKILL) -> None:
-        self.run(
+        result = self.run(
             f"-{signum} {pid}",
             shell=True,
             sudo=True,
             force_run=True,
-            expected_exit_code=0,
-            expected_exit_code_failure_message="fail to run "
-            f"{self.command} -{signum} {pid}",
         )
+
+        if result.exit_code != 0:
```

Comment:
There's a wrinkle here where `kill` can be either a shell builtin or an executable and the output isn't guaranteed to be consistent. I suppose we will find out which systems don't give us this output once we start using it, but it might be better to add a test case for `kill` to `core` to verify this kill tool behaves as we expect and add that test to the tier 1 tests.

We will find out immediately which distros are our edge cases and be able to identify new distros with issues without having to dig through other test results.

Reply:
Is there a way to detect this? For example, does `which kill` help?

Reply:
We can add a flag, like ignore_error. When it's true, ignore exit code. If false, raise exception, if it's failed.

Reply:
I think which kill can help, but I'm more concerned about relying on stdout spew.
I think it's fine to merge this but we should also add test to identify platforms which don't use this output spew. 
We can do a little work to make a test to check that this works on all platforms OR we can spend a bunch of time looking at why a bunch of tests failed one by one, does that make sense?

Reply:
As discussed, let's merge it to remove current regressions in test. We will add a test to check platforms which don't throw out this error.